weaver/wps_restapi/jobs/jobs.py

```diff
@@ -582,11 +586,97 @@ def get_job_outputs(request):
     schema = get_job_io_schema_query(request.params.get("schema"), default=JobInputsOutputsSchema.OGC)
     results, _ = get_results(job, request, schema=schema, link_references=False)
     outputs = {"outputs": results}
-    outputs.update({"links": job.links(request, self_link="outputs")})
+    links = job.links(request, self_link="outputs")
+    outputs.update({"links": links})
     outputs = sd.JobOutputsBody().deserialize(outputs)
     return HTTPOk(json=outputs)
 
 
+@sd.provider_result_value_service.get(
+    tags=[sd.TAG_JOBS, sd.TAG_RESULTS, sd.TAG_PROVIDERS],
+    schema=sd.ProviderResultValueEndpoint(),
+    response_schemas=sd.get_prov_result_responses
+)
+@sd.process_result_value_service.get(
+    tags=[sd.TAG_JOBS, sd.TAG_RESULTS, sd.TAG_PROCESSES],
+    schema=sd.ProcessResultValueEndpoint(),
+    response_schemas=sd.get_proc_result_responses
+)
+@sd.job_result_value_service.get(
+    tags=[sd.TAG_JOBS, sd.TAG_RESULTS],
+    schema=sd.JobResultValueEndpoint(),
+    response_schemas=sd.get_job_result_responses
+)
+@sd.provider_output_service.get(
+    tags=[sd.TAG_JOBS, sd.TAG_RESULTS, sd.TAG_PROCESSES],
+    schema=sd.ProviderAnyOutputEndpoint(),
+    response_schemas=sd.get_prov_result_responses
+)
+@sd.process_output_service.get(
+    tags=[sd.TAG_JOBS, sd.TAG_RESULTS, sd.TAG_PROCESSES],
+    schema=sd.ProcessAnyOutputEndpoint(),
+    response_schemas=sd.get_proc_result_responses
+)
+@sd.job_output_service.get(
+    tags=[sd.TAG_JOBS, sd.TAG_RESULTS, sd.TAG_PROCESSES],
+    schema=sd.JobAnyOutputEndpoint(),
+    response_schemas=sd.get_job_result_responses
+)
+@log_unhandled_exceptions(logger=LOGGER, message=sd.InternalServerErrorResponseSchema.description)
+def get_job_output(request):
+    # type: (PyramidRequest) -> AnyResponseType
+    """
+    Retrieve a specific output from a job execution.
+    """
+    job = get_job(request)
+    raise_job_dismissed(job, request)
+    raise_job_bad_status_success(job, request)
+    settings = get_settings(request)
+    output_id = request.matchdict.get("output_id")
+    # Get requested media-type. "*/*" if omit
+    accept = str(request.accept) if request.accept else "*/*"
+    headers = request.headers
+    results = [o for o in job.results if str(o["identifier"]) == output_id]
+    if results:
+        result = results[0]
+        mime_type = get_field(result, "mime_type", search_variations=True, default="")
+        possible_media_types = transform.CONVERSION_DICT.get(mime_type, [])
+        possible_media_types.append(mime_type)
+    else:
+        raise HTTPNotFound(
+            json={
+                "code": "",
+                "description": "The requested output Id is not available in the job results.",
+                "cause": "The output ID is not available",
+                "error": "",
+                "value": ""
+            }
+        )
```

Comment:
It is confusing that one branch deals with transform media-types, while the other deals with the presence of output and HTTTP error.  

Instead, do the `output_id` check against job results by itself (right after `output_id = request.matchdict.get("output_id")` is obtained), and the rest of the function can assume it is resolved.